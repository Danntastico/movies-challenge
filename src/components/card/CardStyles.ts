import { Row, Col, Card } from 'react-bootstrap';
import styled from 'styled-components';

export const RowStyled = styled(Row)`
    margin: 8px 40px 20px 40px;

    @media screen and (max-width: 768px) {
        margin: 8px 30px 20px 30px;
    }
`;

export const ColStyled = styled(Col)`
`;

export const CardStyled = styled(Card)`
    cursor       : pointer;
    margin-top   : 15px;
    margin-bottom: 15px;
    border: 1px solid #d8a200;
    border-radius: 20px;
    box-shadow: 0 0.5px 5px 0 #d8a200, 0 0.5px 5px 0 #d8a200;
    background-image: url('https://picsum.photos/600/300/?random');

    &:hover {
        transform : scale(1.03);
        transition: transform .4s ease 0s;
        box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
    }

    &::before, &::after {
        position  : absolute;
        top       : 0;
        right     : 0;
        bottom    : 0;
        left      : 0;
        transform : scale3d(0, 0, 1);
        transition: transform .1s ease 0s;
        background: rgba(255, 255, 255, 0.03);
        content   : '';
    }

    &::before {
    transform-origin: left top;
    }

    &::after {
        transform-origin: right bottom;
    }

    &:hover::before, &:hover::after, &:focus::before, &:focus::after {
    transform: scale3d(1, 1, 1);
    }
`;

export const CardBody = styled(Card.Body)`
    width: 100%;
    color: #ffff;
    border-radius: 20px;
    padding: 0;

    background: rgba(254,254,254,0);
    background: -moz-linear-gradient(top, rgba(254,254,254,0) 0%, rgba(70,70,70,0) 62%, rgba(52,52,52,0.85) 68%, rgba(28,28,28,0.85) 76%, rgba(0,0,0,0.85) 100%);
    background: -webkit-gradient(left top, left bottom, color-stop(0%, rgba(254,254,254,0)), color-stop(62%, rgba(70,70,70,0)), color-stop(68%, rgba(52,52,52,0.85)), color-stop(76%, rgba(28,28,28,0.85)), color-stop(100%, rgba(0,0,0,0.85)));
    background: -webkit-linear-gradient(top, rgba(254,254,254,0) 0%, rgba(70,70,70,0) 62%, rgba(52,52,52,0.85) 68%, rgba(28,28,28,0.85) 76%, rgba(0,0,0,0.85) 100%);
    background: -o-linear-gradient(top, rgba(254,254,254,0) 0%, rgba(70,70,70,0) 62%, rgba(52,52,52,0.85) 68%, rgba(28,28,28,0.85) 76%, rgba(0,0,0,0.85) 100%);
    background: -ms-linear-gradient(top, rgba(254,254,254,0) 0%, rgba(70,70,70,0) 62%, rgba(52,52,52,0.85) 68%, rgba(28,28,28,0.85) 76%, rgba(0,0,0,0.85) 100%);
    background: linear-gradient(to bottom, rgba(254,254,254,0) 0%, rgba(70,70,70,0) 62%, rgba(52,52,52,0.85) 68%, rgba(28,28,28,0.85) 76%, rgba(0,0,0,0.85) 100%);
    filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#fefefe', endColorstr='#000000', GradientType=0 );

`;

export const CardTitle = styled(Card.Title)`
    font-family: 'Poppins', sans-serif;
    font-weight: 500;
    font-size  : 25px;
    width      : 35px;
    padding-left: 12px;
    margin-bottom: 18vh;
    border-top-left-radius: 20px 20px;
    border-bottom-right-radius: 20px 20px;
    border: 1px solid #d8a200;
    background-color: rgba(49, 44, 25, 0.7);

    @media screen and (max-width: 768px) {
        margin-bottom: 150px;
    }
`;

export const CardRating = styled(Card.Text)`
    font-family: 'Poppins', sans-serif;
    font-weight: 500;
    font-size  : 19px;
    width      : 50px;
    margin-left: 75%;
    color: #f6be0e;
    background-color: rgba(38, 38, 38, 0.75);
    padding: 10px 5px 10px 5px;
    border-radius: 10px;
    border: 1px solid #d8a200;
    text-align: center;
`;

export const CardText = styled(Card.Text)`
    font-family: 'Poppins', sans-serif;
    font-weight: 500;
    font-size  : 19px;
    width      : 85%;
    margin: auto;

    
    @media screen and (max-width: 768px) {
        font-size  : 15px;
    }
`;

export const CardFooter = styled(Card.Footer)`
    width: 80%;
    margin: auto;
    border-top: 1px solid #69614C;
    font-family: 'Poppins', sans-serif;
    font-weight: 500;
    font-size  : 19px;
    color: #225a7d;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-between;
    align-content: stretch;
    align-items: center;
`;

export const CardParagraph = styled(Card.Text)`
    font-family: 'Poppins', sans-serif;
    font-weight: 500;
    font-size  : 10px;
    width      : 40%;
    padding: 4px 0px 4px 0px;
    margin: 0;
    border-radius: 10px;
    background-color: #212121;
    order: 0;
    flex: 0 1 auto;
    align-self: auto;
    text-align: center;
`;


