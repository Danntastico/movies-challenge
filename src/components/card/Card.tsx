import React, { FunctionComponent } from 'react';
import { ColStyled, CardStyled, CardBody, CardTitle, CardRating, CardText, CardFooter } from './CardStyles';

interface CardProps {
    keyValue: string;
    rating: string;
    title: string;
}

export const Card: FunctionComponent<CardProps> = ({keyValue, rating, title, children}) => {
    return (
        <>
            <ColStyled sm={6} md={4} lg={3} className="align-items-stretch">
                <CardStyled>    
                    <CardBody >
                        <CardTitle> { keyValue } </CardTitle>
                        <CardRating> { rating } </CardRating>
                        <CardText> { title } </CardText>
                        <CardFooter>                                    
                            {children}
                        </CardFooter> 
                    </CardBody>
                </CardStyled>
            </ColStyled>
        </>
    )
}