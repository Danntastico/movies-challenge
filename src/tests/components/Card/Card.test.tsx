import React from 'react'
import { Card } from 'components/card/Card';
import { mount } from 'enzyme';

import '@testing-library/jest-dom'
describe('Unit Testing to CARD component', () => {
    const shall = 
        mount(
            <Card 
                title="testTitle" 
                rating={1} 
                keyValue={1} 
                tmdbID={1} 
                > 
                category 
            </Card>)

    test('Should render correctly', () => {
        expect(shall).toMatchSnapshot();    
    });
    test('Should show title correctly', () => {
        const paragraphs = shall.find('p');
        expect(paragraphs.at(1).text().trim()).toBe('testTitle');
    });
});