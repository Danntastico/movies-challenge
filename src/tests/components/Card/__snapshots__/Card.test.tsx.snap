// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Testing to CARD component Should render correctly 1`] = `
<div
  className="sc-gtssRu gqeTD align-items-stretch col-lg-3 col-md-4 col-sm-6"
>
  <div
    className="sc-dlnjPT jvwfUd card"
  >
    <div
      className="sc-hKFyIo bIulTT card-body"
    >
      <div
        className="sc-eCApGN jjpycP card-title h5"
      >
         
        1
         
      </div>
      <p
        className="sc-jSFkmK EveGJ card-text"
      >
         
        1
         
      </p>
      <p
        className="sc-gKAblj keoKAQ card-text"
      >
         
        testTitle
         
      </p>
      <div
        className="sc-iCoHVE kcKcLJ card-footer"
      />
    </div>
  </div>
</div>
`;
